src/test/java/com/hubspot/jinjava/lib/filter/RootFilterTest.java

```diff
@@ -0,0 +1,68 @@
+package com.hubspot.jinjava.lib.filter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.math.BigDecimal;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.hubspot.jinjava.Jinjava;
+import com.hubspot.jinjava.interpret.InvalidArgumentException;
+import com.hubspot.jinjava.interpret.InvalidInputException;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+
+public class RootFilterTest {
+
+  JinjavaInterpreter interpreter;
+  RootFilter filter;
+
+  @Before
+  public void setup() {
+    interpreter = new Jinjava().newInterpreter();
+    filter = new RootFilter();
+  }
+
+  @Test
+  public void itCalculatesSquareRoot() {
+    assertThat(filter.filter(100, interpreter)).isEqualTo(10D);
+    assertThat(filter.filter(22500, interpreter)).isEqualTo(150D);
+  }
+
+
+  @Test
+  public void itCalculatesCubeRoot() {
+    assertThat(filter.filter(125, interpreter, "3")).isEqualTo(5d);
+  }
+
+  @Test
+  public void itCalculatesNthRoot() {
+    assertThat(filter.filter(9765625d, interpreter, "5.0")).isEqualTo(25d);
+  }
+
+  @Test
+  public void itCalculatesNthRootOfBigDecimal() {
+    BigDecimal result = ((BigDecimal) filter.filter(new BigDecimal(9765625d), interpreter, "5.0"));
```

Comment:
can this be computationally expensive with large values?

Reply:
Performance is largely a function of precision in this case set to 50 digits. With such a low precision operations are in the nanoseconds range. The goal of this library we are using is for performance remain stable as the input grows so we should be safe http://obermuhlner.ch/wordpress/2016/06/02/bigdecimalmath/